flintrock/ec2.py

```diff
@@ -57,33 +58,47 @@ def __init__(
             vpc_id: str,
             master_instance: 'boto3.resources.factory.ec2.Instance',
             slave_instances: "List[boto3.resources.factory.ec2.Instance]",
+            use_private_vpc,
```

Comment:
The plan is then to add an attribute that keeps the subnet_id. Do we include a subnet_is_private method that queries the subnet to determine if it private? In the case we make this change, get_cluster, get_clusters and _compose_cluster also need the subnet_id.